common/map_test.go

```diff
@@ -0,0 +1,188 @@
+package common
+
+import (
+	"slices"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+//nolint:funlen,gocognit, cyclop
+func TestGenericMap(t *testing.T) {
+	t.Parallel()
+
+	//nolint:varnamelen
+	m := GenericMap{
+		"string": "string",
+		"int":    1,
+		"float":  1.1,
+		"map":    map[string]any{"key": "value"},
+		"bool":   true,
+	}
+
+	t.Run("Get", func(t *testing.T) {
+		t.Parallel()
+
+		want := "string"
+		got, err := m.Get("string")
+
+		require.NoError(t, err)
+		assert.Equal(t, want, got)
+
+		var want2 any = nil
+
+		got2, err2 := m.Get("notfound")
+
+		require.Error(t, err2)
+		assert.Equal(t, want2, got2)
+
+		want3 := map[string]any{"key": "value"}
+		got3, err3 := m.Get("map")
+
+		require.NoError(t, err3)
+		assert.Equal(t, want3, got3)
+
+		passes := []string{"string", "int", "float", "map", "bool"}
+
+		for k := range m {
+			got, err := m.Get(k)
+			if !slices.Contains(passes, k) {
+				require.Error(t, err)
+				assert.Nil(t, got)
+			}
+		}
+	})
+
+	t.Run("GetInt", func(t *testing.T) {
+		t.Parallel()
+
+		want := int64(1)
+		got, err := m.GetInt("int")
+
+		require.NoError(t, err)
+		assert.Equal(t, want, got)
+
+		passes := []string{"int"}
```

Comment:
can you rename `passes` something like `keysWithIntValue` for clarity? Similarly for the other test cases in this file.